### PR TITLE
feat(experiments): ability to specify concurrency in run_experiment and evaluate_experiment

### PR DESCRIPTION
**Pull request by anton164**
(Arize-AI#4186)

### CODE REVIEW

**Comment by aabilov-dataminr** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by anton164** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by anton164** [comment]
This MR might be misguided as I'm not really observing any concurrency in the current implementation 🤔 .. Seems like tasks are always evaluated in sequence? Curious to get your guidance

**Comment by anticorrelator** [comment]
Hi @anton164, this PR does correctly wire up configuring the concurrency of our experiment runners—*if the tasks that are being run are async*. We generate a sequence of tasks to run (concurrently if possible) then submit them to our executor.

Would it be possible to rewrite your task as a coroutine function instead and report back if the default level of concurrency works for you?

**Comment by anton164** [comment]
Thanks @anticorrelator -- concurrency works in my experiments with a coroutine task, but the level of concurrency is not sufficient. I’d like to use ~20 concurrency for my batch experiments.

I’ve updated the PR to fix a typo and add docs about the task needing to be a coroutine

File: src/phoenix/experiments/functions.py [hunk 1]
```diff
@@ -158,6 +159,9 @@ def run_experiment(
             examples of the given size. Defaults to False.
         print_summary (bool): Whether to print a summary of the experiment and evaluation results.
             Defaults to True.
+        concurrency (int): concurrency for task execution. The task must be an async coroutine
```
Comment (anticorrelator) [inline src/phoenix/experiments/functions.py:162]: Can we change this to:

```
Specifies the concurrency for task execution. In order to enable concurrent task execution, the task callable must be a coroutine function. Defaults to 3.
```

Comment (anton164) [reply src/phoenix/experiments/functions.py]: Updated

**Review by anticorrelator — COMMENTED** [review]
Thanks so much for the contribution @anton164!

To fully wire this up we need to add a one more thing:

On line 416 we should pass the concurrency parameter to `evaluate_experiment`.

Please let me know if setting the concurrency alone gives you the desired throughput. There are more things you can adjust if you run into throughput issues while using LLM evaluators too.

**Comment by anton164** [comment]
> To fully wire this up we need to add a one more thing:
> 
> On line 416 we should pass the concurrency parameter to `evaluate_experiment`.

Good catch @anticorrelator, updated! Thanks for the quick reviews

File: src/phoenix/experiments/functions.py [hunk 2]
```diff
@@ -427,6 +433,7 @@ def evaluate_experiment(
     dry_run: Union[bool, int] = False,
     print_summary: bool = True,
     rate_limit_errors: Optional[RateLimitErrors] = None,
+    concurrency: int = 3
```
Comment (mikeldking) [inline src/phoenix/experiments/functions.py:436]:
```suggestion
    concurrency: int = 3,
```
Caught by our linter

**Comment by mikeldking** [comment]
Thanks for your contribution @anton164 ! 💯

File: src/phoenix/experiments/functions.py (hunk 2 above)
Comment (anton164) [reply src/phoenix/experiments/functions.py]: addressed

**Review by anticorrelator — APPROVED** [review]
LGTM!

**Comment by anticorrelator** [comment]
@anton164 Looks like there's a line length linting issue in the docstring before we can merge

**Comment by anton164** [comment]
@anticorrelator argh, sorry about that. Hadn't set up linting in my dev environment. I set it up now and confirmed that linting for this file no longer fails after fixing the line ending.

**Pull request merged**